flink-python/pyflink/datastream/connectors/elasticsearch.py

```diff
@@ -0,0 +1,255 @@
+################################################################################
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+import abc
+from enum import Enum
+from typing import List
+
+from pyflink.datastream.connectors import Sink, DeliveryGuarantee
+from pyflink.java_gateway import get_gateway
+from pyflink.util.java_utils import to_jarray, load_java_class
+
+
+class FlushBackoffType(Enum):
+    """
+    Used to control whether the sink should retry failed requests at all or with which kind back off
+    strategy.
+
+    :data: `CONSTANT`:
+
+    After every failure, it waits a configured time until the retries are exhausted.
+
+    :data: `EXPONENTIAL`:
+
+    After every failure, it waits initially the configured time and increases the waiting time
+    exponentially until the retries are exhausted.
+
+    :data: `NONE`:
+
+    The failure is not retried.
+    """
+
+    CONSTANT = 0,
+    EXPONENTIAL = 1,
+    NONE = 2,
+
+    def _to_j_flush_backoff_type(self):
+        JFlushBackoffType = get_gateway().jvm \
+            .org.apache.flink.connector.elasticsearch.sink.FlushBackoffType
+        return getattr(JFlushBackoffType, self.name)
+
+
+class ElasticsearchSinkBuilderBase(abc.ABC):
+    """
+    Base builder to construct a ElasticsearchSink.
+    """
+
+    @abc.abstractmethod
+    def __init__(self):
+        self._j_elasticsearch_sink_builder = None
+
+    def set_emitter(self, emitter_class_name: str):
```

Comment:
This is not usable for Python users at all. There is no default implementations available for use. It means that Python users will have to write Java code to use ES connector.

Reply:
I implement a `PythonSimpleElasticsearchEmiiter`, I have tested it insert and update is successful.